src/nxt_php_sapi.c

```diff
@@ -411,10 +422,63 @@ nxt_php_setup(nxt_task_t *task, nxt_process_t *process,
         nxt_php_set_options(task, value, ZEND_INI_USER);
     }
 
+    zend_hash_init(&nxt_php_user_config_cache, 0, NULL, (dtor_func_t) nxt_php_user_config_cache_entry_dtor, 1);
+
     return NXT_OK;
 }
 
 
+static void
+nxt_php_activate_user_config(nxt_php_run_ctx_t ctx) {
+    char                     *ptr;
+    user_config_cache_entry  *new_entry, *entry;
+    nxt_time_t               request_time = (nxt_time_t) sapi_get_request_time();
+    char                     path[ctx.script_filename.length + 1];
```

Comment:
It is not allowed to use variable length arrays.
Assignment in a declaration is forbidden.
Type names should be sorted by type name length.